rmw_cyclonedds_cpp/CMakeLists.txt

```diff
@@ -22,7 +22,7 @@ if(NOT CMAKE_CXX_STANDARD)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra -Wpedantic)
+  add_compile_options(-Wall -Wextra -Wpedantic -Wimplicit-fallthrough)
```

Comment:
why not disabling the warning all together with `-Wno-implicit-fallthrough` or `-Wimplicit-fallthrough=0`

Reply:
Because the goal is not to remove fallthrough - it's to make sure wherever we us it, it's intentional and obvious.